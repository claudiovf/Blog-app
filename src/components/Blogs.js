import React from 'react'
import { connect } from 'react-redux'
import { removeBlog, updateBlog } from '../reducers/blogReducer'
import { Link } from 'react-router-dom'
import { Table } from 'react-bootstrap'



const Blogs = (props) => {

	return (
		<div className="blogDiv">
			<h2>Blogs</h2>
			<Table>
				<tbody>
					{props.blogs.map((blog, i) =>
						<tr>
							<td>
								<Link key={i} to={`/blogs/${blog.id}`}>
									{blog.title}
								</Link>
							</td>
							<td>
								{blog.likes} Likes
							</td>
						</tr>
					)}
				</tbody>
			</Table>
		</div>
	)
}

const mapStateToProps = state => {
	return { blogs: state.blogs}
}

const ConnectedBlogs = connect(
	mapStateToProps,
	{removeBlog, updateBlog}
)(Blogs)
export default ConnectedBlogs
