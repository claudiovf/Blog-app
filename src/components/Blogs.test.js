import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Blogs from './Blogs'


describe('<Blogs /> ', () => {
	let component
	const blogs = [{
		title: 'Testing Blog Script',
		author: 'Jest',
		url: 'www.bbc.com',
		likes: 30,
		id: 123456
	}]


	const tree = renderer
		.create(<Blogs blogs={blogs} />)
		.toJSON()

	expect(tree).toMatchSnapshot()

	beforeEach(() => {
		component = render(
			<Blogs blogs={blogs} />
		)

	})


	test('displays only Blog Title <Togglable />  by default', () => {
		const blog = component.container.querySelector('.togglableContent')

		expect(blog.textContent).toContain('Testing Blog Script')
		expect(blog.textContent).not.toContain('author')
		expect(blog.textContent).not.toContain('url')
		expect(blog.textContent).not.toContain('likes')

	})

	test('displays Title and Author, but not Likes and URL', () => {
		const button = component.getByText('view')
		fireEvent.click(button)

		const blog = component.container.querySelector('.visibleTogglable')

		expect(blog.textContent).toContain('Testing Blog Script')
		expect(blog.textContent).toContain('Author')
		expect(blog.textContent).toContain('URL')
		expect(blog.textContent).toContain('Likes')

	})

})