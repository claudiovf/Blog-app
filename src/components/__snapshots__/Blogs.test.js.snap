// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <h2>
    Blogs
  </h2>
  <div
    className="blogOnList"
  >
    <div>
      <div
        className="togglableContent"
        style={
          Object {
            "display": "",
          }
        }
      >
        Testing Blog Script
         
        <button
          onClick={[Function]}
        >
           
          view
           
        </button>
      </div>
      <div
        className="visibleTogglable"
        style={
          Object {
            "display": "none",
          }
        }
      >
        Testing Blog Script
         
        <button
          onClick={[Function]}
        >
           
          hide
           
        </button>
        <div>
          URL: 
          www.bbc.com
        </div>
        <div>
          Likes: 
          30
          <button
            onClick={[Function]}
          >
            like
          </button>
        </div>
        <div>
          Author: 
          Jest
        </div>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
